src/api/utils/auth0utils.py

```
@@ -94,8 +95,14 @@ def get_token(self):
             token = get_token.client_credentials(self.client_id, self.client_secret, audience)
             mgmt_api_token = token["access_token"]
             return mgmt_api_token
+        except Auth0Error as e:
+            if e.status_code == 429:
+                logger.warning("[auth0.rate_limit] Management API rate limit hit: %s", e)
+            else:
+                logger.error("[auth0.service_unavailable] Management API error status=%s: %s", e.status_code, e)
+            raise Auth0ServiceUnavailable()
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)